drivers/ethernet/eth_stm32_hal.c

```diff
@@ -28,22 +28,137 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+#define ETH_TXBUFNB                     ETH_TX_DESC_CNT
+
+#define ETH_MEDIA_INTERFACE_MII         HAL_ETH_MII_MODE
+#define ETH_MEDIA_INTERFACE_RMII        HAL_ETH_RMII_MODE
+
+#define LL_AHB1_GRP1_PERIPH_ETHMAC      LL_AHB1_GRP1_PERIPH_ETH1MAC
+#define LL_AHB1_GRP1_PERIPH_ETHMACTX    LL_AHB1_GRP1_PERIPH_ETH1TX
+#define LL_AHB1_GRP1_PERIPH_ETHMACRX    LL_AHB1_GRP1_PERIPH_ETH1RX
+#define LL_AHB1_GRP1_PERIPH_ETHMACPTP   LL_AHB1_GRP1_PERIPH_ETH1MAC
+
+#define ETH_DMA_TRANSMIT_TIMEOUT        20U
+
+static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef *heth,
+											   uint32_t PHYAddr,
+											   uint32_t PHYReg,
+											   uint32_t *RegVal)
+{
+	return HAL_ETH_ReadPHYRegister(heth, PHYAddr, PHYReg, RegVal);
+}
+#else  /* CONFIG_SOC_SERIES_STM32H7X */
+/**
+ * Definitions for ethernet driver of all others STM32 except STM32H7
+ */
+
+#define GET_FIRST_DMA_TX_DESC(heth)  ((ETH_DMADescTypeDef *) \
+										(((ETH_HandleTypeDef *)heth)->TxDesc))
+#define GET_FIRST_DMA_RX_DESC(heth)  ((ETH_DMADescTypeDef *) \
+										(((ETH_HandleTypeDef *)heth)->RxDesc))
+
+#define GET_CURRENT_DMA_TX_DESC(heth)    ((ETH_DMADescTypeDef *) \
+										  (((ETH_HandleTypeDef *)heth)->TxDesc))
+
+#define IS_ETH_DMATXDESC_OWN(dma_tx_desc)    (((ETH_DMADescTypeDef *) \
+												dma_tx_desc)->Status & \
+													ETH_DMATXDESC_OWN)
+#define GET_DMA_TX_BUFFER1_ADDR(dma_tx_desc) (((ETH_DMADescTypeDef *) \
+												dma_tx_desc)->Buffer1Addr)
+
+static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef *heth,
+											   uint32_t PHYAddr,
+											   uint32_t PHYReg,
+											   uint32_t *RegVal)
+{
+	ARG_UNUSED(PHYAddr);
+	return HAL_ETH_ReadPHYRegister(heth, PHYReg, RegVal);
+}
+
+#endif  /* CONFIG_SOC_SERIES_STM32H7X */
+
 #if defined(CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER) && \
-	    DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
-static ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] __dtcm_noinit_section;
-static ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] __dtcm_noinit_section;
-static uint8_t dma_rx_buffer[ETH_RXBUFNB][ETH_RX_BUF_SIZE] __dtcm_noinit_section;
-static uint8_t dma_tx_buffer[ETH_TXBUFNB][ETH_TX_BUF_SIZE] __dtcm_noinit_section;
+			DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
+
+#define ETH_DMA_MEM_SECTION     __dtcm_noinit_section
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+#error "Error: CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER is not applicabble \
+		for STM32H7 series in this module!"
+#endif
+
+#else
+
+#define ETH_DMA_MEM_SECTION     __aligned(4)
+
+#endif
+
+#if defined(CONFIG_NOCACHE_MEMORY)
+#define NO_CACHE_DATA __nocache
 #else
-static ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] __aligned(4);
-static ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] __aligned(4);
-static uint8_t dma_rx_buffer[ETH_RXBUFNB][ETH_RX_BUF_SIZE] __aligned(4);
-static uint8_t dma_tx_buffer[ETH_TXBUFNB][ETH_TX_BUF_SIZE] __aligned(4);
-#endif /* CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER */
```

Comment:
When making changes that apply to other series, this should be explained and put in dedicated commit.
We need to ensure the existing code is not broken or understand there are good reasons to change it.
In any case, don't modify it silently.

Reply:
Comes with next push.